pkg/es/config/config.go

```diff
@@ -585,3 +608,198 @@ func (c *Configuration) Validate() error {
 	_, err := govalidator.ValidateStruct(c)
 	return err
 }
+
+// Run starts the cleaner that runs periodically based on the Frequency field.
+func (cfg *Configuration) RunCleaner() {
```

Comment:
config package is for configuration, not for business logic. This should be somewhere in plugin/storage/es